src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -749,7 +749,8 @@ protected virtual void SaveSourceEtag(DocumentsOperationContext context)
 
             private void RecordDatabaseChangeVector(DocumentsOperationContext context)
             {
-                if (context.DocumentDatabase.ReplicationLoader.IncomingHandlers.Any() == false)
+                var handlers = context.DocumentDatabase.ReplicationLoader.IncomingHandlers;
```

Comment:
maybe a better option will be to pass this handler to this method? I think you can attach the it to the `DataForReplicationCommand` object

Reply:
done